base/set.jl

```diff
@@ -3,13 +3,21 @@
 struct Set{T} <: AbstractSet{T}
     dict::Dict{T,Nothing}
 
-    Set{T}() where {T} = new(Dict{T,Nothing}())
-    Set{T}(s::Set{T}) where {T} = new(Dict{T,Nothing}(s.dict))
+    global unsafe_set(dict::Dict{T,Nothing}) where {T} = new{T}(dict)
```

Comment:
Calling this `unsafe_set` seems unnecessary, since I can't think of any way this would be unsafe. Do we have any precedent for this pattern? It seems common enough and simple, but I just can't think of something similar, which didn't just use inner and outer constructors. Perhaps name it `_internal_wrap_set` or `as_set`?

Reply:
I also think this isn’t really unsafe and I’m open to renaming it. The only thing that could be viewed as “unsafe” is that the resulting `Set` shares memory with the `Dict` that is passed as an argument, which is never the case with the other `Set` constructors.

However, we also have `unsafe_rational` for creating a `Rational` without calling `gcd`, so there is a precedent.

Reply:
I think we can make this `global _Set(dict::Dict{T,nothing}) where {T}`, and everything else looks good, so ready to merge after that

Reply:
Okay, I have renamed it.